### PR TITLE
changes to align with spec

### PR DESCRIPTION
**Pull request by MilesHolland**
A collection of small changes to the ECI and IP evaluators to match the spec more closely. Changes include:
- refactor non-content safety evaluator response parsing to add metric prefix to results. This includes special logic to capitalize the 'eci' prefix.
- Changed ECI simulator enum to just be 'ECI'.

### CODE REVIEW

**Review by diondrapeck — DISMISSED** [review]

File: src/promptflow-evals/promptflow/evals/evaluators/_eci/_eci.py [hunk 1]
```diff
@@ -65,8 +65,8 @@ class ECIEvaluator:
     .. code-block:: python
 
         {
-            "label": "False",
-            "reasoning": "Some reason."
+            "ECI_score": "False",
```
Comment (luigiw) [inline src/promptflow-evals/promptflow/evals/evaluators/_eci/_eci.py:68]: Hmmm, we call it a score, but the value is boolean?

Comment (MilesHolland) [reply src/promptflow-evals/promptflow/evals/evaluators/_eci/_eci.py:68]: This is aligning with the spec, I'm neutral on changing this - on one hand. It's consistent with other evaluators. On the other hand, it's odd. Minsoo was also neutral on this.

File: src/promptflow-evals/promptflow/evals/evaluators/_eci/_eci.py [hunk 2]
```diff
@@ -65,8 +65,8 @@ class ECIEvaluator:
...
+            "ECI_reasoning": "Some reason."
```
Comment (singankit) [inline src/promptflow-evals/promptflow/evals/evaluators/_eci/_eci.py:69]: Should this be camel case ?

Comment (MilesHolland) [reply src/promptflow-evals/promptflow/evals/evaluators/_eci/_eci.py:69]: I don't believe so, extant evaluators already return this results with snake-cased keys, this is just following that pattern.

File: src/promptflow-evals/promptflow/evals/evaluators/_eci/_eci.py (hunk 1 above)
Comment (MilesHolland) [reply src/promptflow-evals/promptflow/evals/evaluators/_eci/_eci.py]: Reverted back to label.

File: src/promptflow-evals/promptflow/evals/_common/rai_service.py [hunk 3]
```diff
@@ -216,12 +216,27 @@ def parse_response(  # pylint: disable=too-many-branches,too-many-statements
         response = response.replace("true", "True")
         parsed_response = literal_eval(response)
         result = {}
-        result["label"] = parsed_response["label"] if "label" in parsed_response else np.nan
-        result["reasoning"] = parsed_response["reasoning"] if "reasoning" in parsed_response else ""
+        metric_prefix = _get_metric_prefix(metric_name)
+        # Use label instead of score since these are assumed to be boolean results.
+        result[metric_prefix + "_label"] = parsed_response["label"] if "label" in parsed_response else np.nan
```
Comment (ninghu) [inline src/promptflow-evals/promptflow/evals/_common/rai_service.py:221]: `np.nan` is float type. I wonder if we should just return an empty string.

Comment (MilesHolland) [reply src/promptflow-evals/promptflow/evals/_common/rai_service.py]: Done

**Review by ninghu — APPROVED** [review]

**Review by diondrapeck — APPROVED** [review]

**Pull request merged**